### PR TITLE
AK+Everywhere: Port LexicalPath to new Strings

### PR DESCRIPTION
**Pull request by sppmacd**
Ladybird PR: https://github.com/SerenityOS/ladybird/pull/146

Known things that don't work:
* [x] FileManager asserts when launching (something about model row count)
* [x] Memory leak in LexicalPath::from_string() somehow (see [CI](https://dev.azure.com/SerenityOS/SerenityOS/_build/results?buildId=23457&view=logs&j=2566f013-1391-5cbe-90ce-ae4ec398ae06&t=84932543-dadf-5f2f-567f-c183cdc02e79)) -> #16385

- AK: Add StringBuilder::try_append() taking new Strings
- AK: Add String::substring_from_byte_offset(start)
- AK: Add String::find* functions
- AK: Port some string manipulation functions from DeprecatedString
- AK: Port LexicalPath to new Strings

### CODE REVIEW

File: AK/String.cpp [hunk 1]
```diff
@@ -309,6 +311,70 @@ ErrorOr<String> String::replace(StringView needle, StringView replacement, Repla
     return StringUtils::replace(*this, needle, replacement, replace_mode);
 }
 
+bool String::contains(String const& needle, CaseSensitivity case_sensitivity) const
```
Comment (linusg) [inline AK/String.cpp:314]: not convinced we need this overload, just taking a string view in all cases seems fine?

File: AK/String.cpp [hunk 2]
```diff
@@ -309,6 +311,70 @@ ErrorOr<String> String::replace(StringView needle, StringView replacement, Repla
...
+{
+    return StringUtils::contains(bytes_as_string_view(), needle.bytes_as_string_view(), case_sensitivity);
+}
+
+bool String::contains(StringView needle, CaseSensitivity case_sensitivity) const
```
Comment (linusg) [inline AK/String.cpp:319]: `contains_bytes`, probably

File: AK/String.cpp [hunk 3]
```diff
@@ -309,6 +311,70 @@ ErrorOr<String> String::replace(StringView needle, StringView replacement, Repla
...
+{
+    return StringUtils::starts_with(bytes_as_string_view(), string, case_sensitivity);
+}
+
+bool String::ends_with_bytes(StringView string, CaseSensitivity case_sensitivity) const
+{
+    return StringUtils::ends_with(bytes_as_string_view(), string, case_sensitivity);
+}
+
+bool String::starts_with(u32 codepoint) const
```
Comment (linusg) [inline AK/String.cpp:334]: Why do we need all of these single code point APIs? Hardly seems useful, but perhaps I'm missing something

**Review by linusg — COMMENTED** [review]
haven't reviewed the last commit

File: AK/String.cpp (hunk 3 above)
Comment (sppmacd) [reply AK/String.cpp:334]: They were implemented in DeprecatedString, but are not used (yet). I will remove them for now.

File: AK/StringBuilder.h [hunk 4]
```diff
@@ -23,6 +23,7 @@ class StringBuilder {
 
     ErrorOr<void> try_append(StringView);
 #ifndef KERNEL
+    ErrorOr<void> try_append(String const&);
```
[the remark this answers is not in the document]
Comment (sppmacd) [reply AK/StringBuilder.h]: seems to work fine, removed.

File: AK/String.cpp (hunk 1 above)
Comment (sppmacd) [reply AK/String.cpp]: fixed

File: AK/String.cpp (hunk 2 above)
Comment (sppmacd) [reply AK/String.cpp]: sure, fixed.

**Comment by ADKaster** [comment]
This currently has conflicts.

**Comment by sppmacd** [comment]
After [this discussion](https://discord.com/channels/830522505605283862/830739873119207426/1057052959277580308) I'm not even sure if this PR makes sense.

**Pull request closed without merging**